exonum/src/api/backends/actix.rs

```diff
@@ -373,16 +378,38 @@ impl SystemRuntime {
             }
             api_runtime_addresses
         };
+        // Starts additional workers.
+        let blockchain = aggregator.blockchain().clone();
+        let additional_workers = aggregator
+            .additional_workers()
+            .into_iter()
+            .map(move |(name, worker)| {
+                let (cancel, context) = ServiceWorkerContext::new(blockchain.clone());
+                let thread_handle = thread::Builder::new()
+                    .name(format!("{}:additional_worker", name))
+                    .spawn(move || worker(context))
+                    .unwrap();
+                (thread_handle, cancel)
+            })
+            .collect();
 
         Ok(Self {
             system_thread,
             system_address,
             api_runtime_addresses,
+            additional_workers,
         })
     }
 
     /// Stops the actix system runtime along with all web runtimes.
     pub fn stop(self) -> result::Result<(), failure::Error> {
+        // Stop all additional workers.
+        for (handle, cancel) in self.additional_workers {
+            drop(cancel);
+            handle
+                .join()
+                .map_err(|_| format_err!("Unable to stop additional worker"))??;
```

Comment:
1. `??` looks weird.
2. Maybe write worker name?
3. Maybe move logic to the separate method `stop_worker`?